packages/wpcom-checkout/src/types.ts

```diff
@@ -33,21 +33,34 @@ export interface TaxVendorInfo {
 	 */
 	country_code: string;
 
-	/**
-	 * The localized name of the tax (eg: "VAT", "GST", etc.).
-	 */
-	tax_name: string;
-
 	/**
 	 * The mailing address to display on receipts as a list of strings (each
 	 * string should be on its own line).
 	 */
 	address: string[];
 
+	/**
+	 * An object containing tax names and corresponding vendor ids that are used for the user's country
+	 *
+	 * This will deprecate the vat_id and tax_name properties
+	 * For now, those two properties will stay in place for backwards compatibility
+	 *
+	 * Property: The localized name of the tax (eg: "VAT", "GST", etc.).
+	 * Value: A8c vendor id for that specific tax
+	 */
+	tax_name_and_vendor_id_array: object;
```

Comment:
It's not a good idea to use the `object` type because it doesn't tell us much about what the object looks like. Here, we know what the object looks like so we can encode that in the type.

```suggestion
	tax_name_and_vendor_id_array: Record< string, string >;
```

The `Record` type is a TypeScript helper that means "an object with key-value pairs". It requires two type arguments (the things in angle brackets) which represent the type of the key, and the type of the value, respectively. In this case, both the key and the value are strings.

Reply:
`TIL`

I think this whole PR is a reminder of just how much I still have to learn about Frontend dev work.